pkg/controller/generic_reconciler_test.go

```diff
@@ -0,0 +1,47 @@
+package controller
+
+import (
+	"testing"
+
+	"github.com/app-sre/dv-operator/pkg/testutils"
+	apps_v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestGenericReconciler(t *testing.T) {
+	deployment, err := testutils.CreateDeploymentFromTemplate(
+		testutils.NewTemplateArgs())
+	if err != nil {
+		t.Errorf("Error creating deployment from template %v", err)
+	}
+
+	// Objects to track in the fake client.
+	objs := []runtime.Object{&deployment}
+	s := scheme.Scheme
+	s.AddKnownTypes(apps_v1.SchemeGroupVersion, &deployment)
+
+	// Create a fake client to mock API calls.
+	client := fake.NewFakeClient(objs...)
+
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{Name: "foo", Namespace: "bar"},
+	}
+
+	gr := &GenericReconciler{
+		client:         client,
+		scheme:         s,
+		reconciledKind: "Deployment",
```

Comment:
Hmm, then should we just remove the `reconciledKind` from the object itself and find the kind in the reconciler code?

Reply:
The reconciledKind is mainly used in log messages, but is determined from the object in the NewGenericReconciler function.

        kind := reflect.TypeOf(obj).String()
        kind = strings.SplitN(kind, ".", 2)[1]

It would be wasteful to go through reflect to get the kind each time we want to log a message with the information.  We don't use it in the Reconcile function either, we instead get the GVK from the object itself.  It probably wouldn't be the end of the world to have a utility function that retrieves the kind from an object in either case.

I would favor keeping the reconciledKind in the GenericReconciler just because we already do a lock with reflect and that isn't very performant.  I don't feel that strongly about it though.  Creating a function to get the kind and having that used by the GenericReconciler and the tests would be good I think.

Reply:
ok, I'll keep it

Reply:
Kept in 0246805